db/migrate/20190902133711_add_tags_links_digest_to_existing_subscriber_lists.rb

```diff
@@ -0,0 +1,10 @@
+class AddTagsLinksDigestToExistingSubscriberLists < ActiveRecord::Migration[5.2]
+  def change
+    SubscriberList.all.each do |list|
+      tags_or_links = list.links.any? ? list.links : list.tags
+      data_type = list.links.any? ? :links : :tags
+      list.tags_links_digest = HashDigest.generate(tags_or_links, data_type)
+      list.save!
+    end
```

Comment:
For this migration it'd probably make sense to disable the transaction as this could gradually lock the whole table. 

If it takes a long time to run you could set it so that it looks up where digests are nil in the table, which would then mean this migration could pick up again if it failed half way through.